web/src/pages/post/[id].tsx

```diff
@@ -52,59 +49,16 @@ export default function Post() {
       },
     });
 
-  const { mutate: mutateLike, isLoading: isLiking } = useMutation({
-    mutationKey: ["like"],
-    mutationFn: () => likePost(postData?.id ?? ""),
-    onError: (err: Error) => {
-      setLocalIsLiked(false); // Revert the local state on failure
-      toast.error(err.message);
-    },
-  });
-
-  const { mutate: mutateUnlike, isLoading: isUnliking } = useMutation({
-    mutationKey: ["unlike"],
-    mutationFn: () => unlikePost(postData?.id ?? ""),
-    onError: (err: Error) => {
-      setLocalIsLiked(true); // Revert the local state on failure
-      toast.error(err.message);
-    },
+  const { mutateLikeChange, likeCount, isLiked } = useLikePost({
+    postId: postData?.id ?? "",
```

Comment:
This feels disingenuous. I would just allow `undefined`/`null` here if it can be empty and then handle that in `useLikePost()`